homeassistant/components/sensor/lyft.py

```diff
@@ -0,0 +1,231 @@
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['lyft_rides==0.1.0b0']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CLIENT_ID = 'client_id'
+CONF_CLIENT_SECRET = 'client_secret'
+CONF_END_LATITUDE = 'end_latitude'
+CONF_END_LONGITUDE = 'end_longitude'
+CONF_PRODUCT_IDS = 'product_ids'
+CONF_START_LATITUDE = 'start_latitude'
+CONF_START_LONGITUDE = 'start_longitude'
+
+ICON = 'mdi:taxi'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_CLIENT_ID): cv.string,
+    vol.Required(CONF_CLIENT_SECRET): cv.string,
+    vol.Required(CONF_START_LATITUDE): cv.latitude,
+    vol.Required(CONF_START_LONGITUDE): cv.longitude,
+    vol.Optional(CONF_END_LATITUDE): cv.latitude,
+    vol.Optional(CONF_END_LONGITUDE): cv.longitude,
+    vol.Optional(CONF_PRODUCT_IDS, default=None):
+        vol.All(cv.ensure_list, [cv.string]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Lyft sensor."""
+    from lyft_rides.auth import ClientCredentialGrant
+
+    auth_flow = ClientCredentialGrant(client_id=config.get(CONF_CLIENT_ID),
+                                      client_secret=config.get(
+                                          CONF_CLIENT_SECRET),
+                                      scopes="public")
+    session = auth_flow.get_session()
+
+    wanted_product_ids = config.get(CONF_PRODUCT_IDS)
+
+    dev = []
+    timeandpriceest = LyftEstimate(
+        session, config[CONF_START_LATITUDE], config[CONF_START_LONGITUDE],
+        config.get(CONF_END_LATITUDE), config.get(CONF_END_LONGITUDE))
+    for product_id, product in timeandpriceest.products.items():
+        if (wanted_product_ids is not None) and \
+           (product_id not in wanted_product_ids):
+            continue
+        dev.append(LyftSensor('time', timeandpriceest, product_id, product))
+        if product.get('estimate') is not None:
+            dev.append(LyftSensor(
+                'price', timeandpriceest, product_id, product))
+    add_devices(dev)
+
+
+class LyftSensor(Entity):
+    """Implementation of an Lyft sensor."""
+
+    def __init__(self, sensorType, products, product_id, product):
+        """Initialize the Lyft sensor."""
+        self.data = products
+        self._product_id = product_id
+        self._product = product
+        self._sensortype = sensorType
+        self._name = '{} {}'.format(self._product['display_name'],
+                                    self._sensortype)
+        if self._sensortype == 'time':
+            self._unit_of_measurement = 'min'
+        elif self._sensortype == 'price':
+            estimate = self._product['estimate']
+            if estimate is not None:
+                self._unit_of_measurement = estimate.get('currency')
+
+        self.update()
```

Comment:
Please don't call `self.update()` but instead pass `True` as second parameter to `add_devices`